packages/_shared/src/hooks/useActiveReleaseLocalesStatuses.ts

```diff
@@ -1,39 +1,44 @@
 import { useCallback, useMemo } from 'react';
 
 import type { ReleaseAction, ReleaseLocalesStatusMap } from '@contentful/field-editor-shared';
-import type { CollectionProp, EntryProps, LocaleProps } from 'contentful-management/types';
+import type {
+  AssetProps,
+  CollectionProp,
+  EntryProps,
+  LocaleProps,
+} from 'contentful-management/types';
 
 import type { ReleaseV2Entity, ReleaseV2EntityWithLocales, ReleaseV2Props } from '../types';
-import { getPreviousReleaseEntryVersion } from '../utils/getPreviousReleaseEntryVersion';
+import { getPreviousReleaseEntity } from '../utils/getPreviousReleaseEntity';
 
 export const useActiveReleaseLocalesStatuses = ({
-  currentEntryDraft,
-  entryId,
+  currentEntityDraft,
+  entityId,
   releaseVersionMap,
   locales,
   activeRelease,
   releases,
 }: {
-  currentEntryDraft: EntryProps;
-  entryId: string;
+  currentEntityDraft: EntryProps | AssetProps;
+  entityId: string;
   releaseVersionMap: Map<string, Map<string, ReleaseAction>>;
   locales: LocaleProps[];
   activeRelease: ReleaseV2Props | undefined;
   releases: CollectionProp<ReleaseV2Props> | undefined;
 }) => {
   const previousReleaseEntity = useMemo(
     () =>
-      getPreviousReleaseEntryVersion({
-        entryId,
+      getPreviousReleaseEntity({
+        entityId,
         releaseVersionMap,
         activeRelease,
         releases,
       }).previousReleaseEntity,
-    [entryId, releaseVersionMap, activeRelease, releases],
+    [entityId, releaseVersionMap, activeRelease, releases],
   );
   const activeReleaseReleaseEntity = useMemo(
-    () => activeRelease?.entities.items.find((entity) => entity.entity.sys.id === entryId),
-    [activeRelease?.entities.items, entryId],
+    () => activeRelease?.entities.items.find((entity) => entity.entity.sys.id === entityId),
```

Comment:
we might also want to check for entity.sys.type (the same id could be theoretically used for entry and asset)

Reply:
Good call, although it also raises the question for me of why we're even passing in the entityId explicitly when we also pass in `currentEntityDraft` (which is a confusing name IMO)